### PR TITLE
RELEASE scheduled on 25/02/2021

### PR DESCRIPTION
**Pull request by yarpo**
Release is scheduled on 25 Feb, 5:30 PM.

**Stories:**
https://tools.hmcts.net/jira/browse/DIV-6448 - Ability to amend court order 
https://tools.hmcts.net/jira/browse/DIV-6578 - General Email
https://tools.hmcts.net/jira/browse/DIV-6450 - Alternative service 
https://tools.hmcts.net/jira/browse/DIV-6553 - General referral

### CODE REVIEW

File: definitions/divorce/json/CaseEvent/CaseEvent-share-a-case-prod.json [hunk 1]
```diff
@@ -55,68 +55,6 @@
     "SecurityClassification": "Public",
     "ShowSummary": "Y"
   },
-  {
```
Comment (yarpo) [inline definitions/divorce/json/CaseEvent/CaseEvent-share-a-case-prod.json:58]: All events are already defined in the core file. They are pretty much the same. The only difference is `preConstionState(s)` - events defined in the core file have more states, but they always have all pre states that we used to have here.

File: definitions/divorce/json/CaseEvent/CaseEvent.json [hunk 2]
```diff
@@ -2958,5 +2958,295 @@
...
+  {
+    "LiveFrom": "09/12/2020",
+    "CaseTypeID": "DIVORCE",
+    "ID": "boWelshReview",
+    "Name": "Welsh Translation Review",
+    "Description": "Welsh Translation Review",
+    "DisplayOrder": 2,
+    "PreConditionState(s)": "AosStarted;AosOverdue;ServiceApplicationNotApproved;AwaitingAlternativeService;AwaitingProcessServerService;AwaitingDWPResponse",
+    "PostConditionState": "WelshResponseAwaitingReview",
+    "SecurityClassification": "Public"
```
Comment (yarpo) [inline definitions/divorce/json/CaseEvent/CaseEvent.json:3174]: definition deleted from RRJ-prod:

```
{
    "LiveFrom": "09/12/2020",
    "CaseTypeID": "DIVORCE",
    "ID": "boWelshReview",
    "Name": "Welsh Translation Review",
    "Description": "Welsh Translation Review",
    "DisplayOrder": 2,
    "PreConditionState(s)": "AosStarted;AosOverdue;ServiceApplicationNotApproved",
    "PostConditionState": "WelshResponseAwaitingReview",
    "SecurityClassification": "Public"
  }
```

File: definitions/divorce/json/CaseEvent/CaseEvent.json [hunk 3]
```diff
@@ -2958,5 +2958,295 @@
...
+    "CaseTypeID": "DIVORCE",
+    "ID": "aosReceivedNoAdConStarted",
+    "Name": "AOS Received (no ad/con)",
+    "Description": "AOS Received (no admission/consent)",
+    "DisplayOrder": 1,
+    "PreConditionState(s)": "AosStarted;AosOverdue;ServiceApplicationNotApproved;AwaitingAlternativeService;AwaitingProcessServerService;AwaitingDWPResponse",
+    "PostConditionState": "AosCompleted",
+    "CallBackURLAboutToSubmitEvent": "${CCD_DEF_COS_URL}/aos-received",
+    "CallBackURLSubmittedEvent": "${CCD_DEF_COS_URL}/aos-submitted",
+    "SecurityClassification": "Public"
```
Comment (yarpo) [inline definitions/divorce/json/CaseEvent/CaseEvent.json:3163]:
```
  {
    "LiveFrom": "15/09/2020",
    "CaseTypeID": "DIVORCE",
    "ID": "aosReceivedNoAdConStarted",
    "Name": "AOS Received (no ad/con)",
    "Description": "AOS Received (no admission/consent)",
    "DisplayOrder": 1,
    "PreConditionState(s)": "AosStarted;AosOverdue;ServiceApplicationNotApproved",
    "PostConditionState": "AosCompleted",
    "CallBackURLAboutToSubmitEvent": "${CCD_DEF_COS_URL}/aos-received",
    "CallBackURLSubmittedEvent": "${CCD_DEF_COS_URL}/aos-submitted",
    "SecurityClassification": "Public"
  },
```

File: definitions/divorce/json/CaseEvent/CaseEvent.json [hunk 4]
```diff
@@ -2958,5 +2958,295 @@
...
+    "CaseTypeID": "DIVORCE",
+    "ID": "aosSubmittedDefended",
+    "Name": "AOS Submitted (defending)",
+    "Description": "AOS Submitted (defending)",
+    "DisplayOrder": 2,
+    "PreConditionState(s)": "AosStarted;AosOverdue;ServiceApplicationNotApproved;AwaitingAlternativeService;AwaitingProcessServerService;AwaitingDWPResponse",
+    "PostConditionState": "AosSubmittedAwaitingAnswer",
+    "CallBackURLAboutToSubmitEvent": "${CCD_DEF_COS_URL}/aos-received",
+    "CallBackURLSubmittedEvent": "${CCD_DEF_COS_URL}/aos-submitted",
+    "SecurityClassification": "Public"
```
Comment (yarpo) [inline definitions/divorce/json/CaseEvent/CaseEvent.json:3150]: definition removed from RRJ-prod:

```
{
    "LiveFrom": "15/09/2020",
    "CaseTypeID": "DIVORCE",
    "ID": "aosSubmittedDefended",
    "Name": "AOS Submitted (defending)",
    "Description": "AOS Submitted (defending)",
    "DisplayOrder": 2,
    "PreConditionState(s)": "AosStarted;AosOverdue;ServiceApplicationNotApproved",
    "PostConditionState": "AosSubmittedAwaitingAnswer",
    "CallBackURLAboutToSubmitEvent": "${CCD_DEF_COS_URL}/aos-received",
    "CallBackURLSubmittedEvent": "${CCD_DEF_COS_URL}/aos-submitted",
    "SecurityClassification": "Public"
  }
```

File: definitions/divorce/json/CaseEvent/CaseEvent.json [hunk 5]
```diff
@@ -2958,5 +2958,295 @@
...
+  {
+    "LiveFrom": "03/12/2020",
+    "CaseTypeID": "DIVORCE",
+    "ID": "startAos",
+    "Name": "AOS started",
+    "Description": "AOS started",
+    "DisplayOrder": 1,
+    "PreConditionState(s)": "AosAwaiting;AwaitingAlternativeService;AwaitingProcessServerService;AwaitingDWPResponse",
+    "PostConditionState": "AosStarted",
+    "SecurityClassification": "Public"
```
Comment (yarpo) [inline definitions/divorce/json/CaseEvent/CaseEvent.json:3079]: removed from RRJ-prod:

```
{
    "LiveFrom": "01/01/2017",
    "CaseTypeID": "DIVORCE",
    "ID": "startAos",
    "Name": "AOS started",
    "Description": "AOS started",
    "DisplayOrder": 1,
    "PreConditionState(s)": "AosAwaiting",
    "PostConditionState": "AosStarted",
    "SecurityClassification": "Public"
  }
```

File: definitions/divorce/json/CaseEvent/CaseEvent.json [hunk 6]
```diff
@@ -2958,5 +2958,295 @@
...
+    "Description": "AOS Submitted (defending)",
+    "DisplayOrder": 2,
+    "PreConditionState(s)": "AosStarted;AosOverdue;ServiceApplicationNotApproved;AwaitingAlternativeService;AwaitingProcessServerService;AwaitingDWPResponse",
+    "PostConditionState": "AosSubmittedAwaitingAnswer",
+    "CallBackURLAboutToSubmitEvent": "${CCD_DEF_COS_URL}/aos-received",
+    "CallBackURLSubmittedEvent": "${CCD_DEF_COS_URL}/aos-submitted",
+    "SecurityClassification": "Public"
+  },
+  {
+    "LiveFrom": "15/09/2020",
```
Comment (jacmurphy) [inline definitions/divorce/json/CaseEvent/CaseEvent.json:3153]: General question. Are we not updating the data of LiveFrom when we alter an event ?

**Review by jacmurphy — APPROVED** [review]

**Review by jvictorcf — APPROVED** [review]
LGTM

File: definitions/divorce/json/CaseEvent/CaseEvent.json (hunk 6 above)
Comment (yarpo) [reply definitions/divorce/json/CaseEvent/CaseEvent.json:3153]: Well, they haven't been updated. They were moved from different file. This is a date when we defined it, not when it was moved to prod.

Obviously it's just an approach. Maybe we will change it one day.

**Pull request merged**